### PR TITLE
n8n automation action integration 

### PR DESCRIPTION
**Pull request by melohagan**
## Description
n8n has been added as an automation action. Supports different http methods, and the 'Authorization' header.

## Addresses
- https://linear.app/budibase/issue/BUDI-4702/n8nio-integration
- https://github.com/Budibase/budibase/issues/1212

## Screenshots
Screenshot 2024-02-08 at 15 45 10

Screenshot 2024-02-08 at 15 44 30

Screenshot 2024-02-08 at 15 44 09

Screenshot 2024-02-08 at 15 45 41

### CODE REVIEW

File: packages/server/src/automations/steps/make.ts
Comment (samwho) [inline packages/server/src/automations/steps/make.ts:58]: How come these're being removed in this PR?

File: packages/server/src/automations/tests/n8n.spec.ts
Comment (samwho) [inline packages/server/src/automations/tests/n8n.spec.ts:14]: test.com is a privately owned domain, I tend to avoid it in tests just in case. It could theoretically point anywhere and if we do accidentally hit it there's no telling what it'll do.

I prefer to use example.com, which is owned by IANA and is a specifically reserved domain: https://www.rfc-editor.org/rfc/rfc2606.

**Review by samwho — CHANGES_REQUESTED** [review]
This automation appears to make a pretty naked `fetch` call, and doesn't appear to have anything n8n-specific in it. Could we not rebrand this as a generic HTTP call step?

File: packages/server/src/automations/steps/make.ts
Comment (melohagan) [reply packages/server/src/automations/steps/make.ts:58]: They were supposed to be fully deprecated in a previous PR and I missed these props at the time. Only noticed today!

https://github.com/Budibase/budibase/commit/b4e0d98973c6395e36040f7789fb2f7995781f10

**Review by shogunpurple — COMMENTED** [review]
Would agree with @samwho here. It's just duplicated code to make an HTTP call to N8N.  I'd say we:
- either maintain this as a plugin and add a tutorial in the docs
- Add something to the docs to show you how to use our REST connector to do it

EDIT: we actually already do this for make and zapier. It's basically the same thing so no argument if we merge this too. We don't have a direct REST step, so you still have to set up a REST datasource in `data`, point it at N8N and then use it in automations. It saves you a few steps if all you want to do is trigger the webhook I guess.

File: packages/server/src/automations/steps/n8n.ts
Comment (mike12345567) [inline packages/server/src/automations/steps/n8n.ts:12]: Could move these into `@budibase/types` and re-use them in the REST integration, just incase these pop up in other places would want to keep from re-writing enums like this all over the place.

Comment (melohagan) [reply packages/server/src/automations/steps/n8n.ts]: Added a HttpMethod enum to types, and updated the REST connector.

**Comment by joebudi** [comment]
@melohagan nice work. Is Authorization an optional setting? If so, we should make that clear.

**Comment by melohagan** [comment]
> @melohagan nice work. Is Authorization an optional setting? If so, we should make that clear.

It is optional, and n8n will return a 401 status and a message requesting the auth header if it's needed.

How would you expect required/optional fields to be presented in automations? This would apply to all automation action steps. E.g. bold labels, or an asterisk beside the label

**Comment by joebudi** [comment]
> > @melohagan nice work. Is Authorization an optional setting? If so, we should make that clear.
> 
> It is optional, and n8n will return a 401 status and a message requesting the auth header if it's needed.
> 
> How would you expect required/optional fields to be presented in automations? This would apply to all automation action steps. E.g. bold labels, or an asterisk beside the label

@AndyWhann is best to answer this.

**Review by samwho — APPROVED** [review]
I'm broadly happy with this change, though I would wait for an OK from Marty or Joe before going ahead and merging.

**Pull request merged**
